frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -707,6 +709,7 @@ function ReaderHighlight:onTap(_, ges)
     local cleared = self.hold_pos and self:clear()
     -- We only care about potential taps on existing highlights, not on taps that closed a highlight menu.
     if not cleared and ges then
+        if #self.ui.annotation.annotations == 0 then return end
```

Comment:
Dunno if little easy bits like these could be shove into a method `ReaderAnnotations:hasAnnotations()`.

Reply:
I believe within the Annotation/Highlight/Bookmark modules we can use `annotation.annotations` directly, kind of small steps to speed-up/reduce memory usage.